### PR TITLE
Task1

### PR DESCRIPTION
**Pull request by jedzer**
### Summary

In this task we should write a server that has:
* GET /auth handler wich sets an auth cookie and sends an HTML with response
* GET /form handler wich returns an HTML with a form and onclick button that sends data to /charge
* POST /charge handler wich returns an {{data}}$ was charged if was auth or 404 otherwise

**Handlers**:
* GET /auth - set cookie auth=True and returns responce
* GET /deauth - set cookie auth=False and returns responce
* GET /form - returns form that submits to /charge
* GET /charge - return message with value got from form if auth otherwise returns 404

### How to Verify 
See README.md

### CODE REVIEW

File: views.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+from simplifiers import template, simple_response
+
+
+def index(request: BaseHTTPRequestHandler):
+    simple_response(request)
+    request.wfile.write(template('index'))
+
+
+def auth(request):
+    request.send_response(202)
```
Comment (andreybulygin) [inline views.py]: typical success code is 200, not 202

File: views.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    raw_cookies = request.headers.get('Cookie')
+    cookies = SimpleCookie(raw_cookies)
+    auth_cookie = cookies.get('auth')
+    auth_message = '<a href="/deauth">De-authenticate</a>'
+    if auth_cookie is None or auth_cookie.value == 'False':
+        auth_message = '<a href="/auth">Authenticate</a>'
+    request.wfile.write(template('form', auth=auth_message))
+
+
+def handle_404(request):
```
Comment (andreybulygin) [inline views.py]: you need the 404 http response here, not the 202

File: views.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    request.send_response(202)
+    request.send_header('content-type', 'text/html')
+    request.send_header('Set-Cookie', 'auth=False; HttpOnly')
+    request.end_headers()
+    request.wfile.write(template('auth', message='De-authenticated'))
+
+
+def charge(request, dollars=None):
+    if dollars is None:
+        return handle_404(request)
```
Comment (andreybulygin) [inline views.py]: good check, but it's 400, not 404. And the best thing here is to also supply the error message (for the user to understand what's wrong)

File: views.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    request.send_header('content-type', 'text/html')
+    request.send_header('Set-Cookie', 'auth=True; HttpOnly')
+    request.end_headers()
+    request.wfile.write(template('auth', message='Authenticated'))
+
+
+def de_auth(request):
+    request.send_response(202)
+    request.send_header('content-type', 'text/html')
+    request.send_header('Set-Cookie', 'auth=False; HttpOnly')
```
Comment (andreybulygin) [inline views.py]: it's better to unset (delete) the cookie at all. Current implementation is also OK, just out of best practice.

File: main.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
+from http.server import HTTPServer
+from http.server import BaseHTTPRequestHandler
+from urllib.parse import parse_qs
+
+from urls import URLS
```
Comment (andreybulygin) [inline main.py]: I'm not sure you want same `URLS` for both POST and GET.
E.g. try `curl -X POST http://localhost:8001/form`

File: views.py (hunk 1 above)
Comment (jedzer) [reply views.py]: I forgot to push new 😅

File: views.py (hunk 2 above)
Comment (jedzer) [reply views.py]: Yup, same as above... Forgot to push. Now it's 404

File: main.py (hunk 5 above)
Comment (jedzer) [reply main.py]: True. I'll change

**Pull request merged**